OpenAPI/OpenAPI.NET/references-openapi.md

```diff
@@ -0,0 +1,665 @@
+components:
+    schemas:
+      item:
+        type: object
+
+```
+
+#### In a subschema of an inline schema
+
+```yaml
+openapi: 3.1.0
+info:
+    title: Reference in at the root of an inline schema
+    version: 1.0.0
+paths:
+    /items:
+        get:
+          responses:
+            200:
+              description: ok
+              content:
+                application/json:
+                  schema:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/item'
+components:
+    schemas:
+      item:
+        type: object
+```
+
+#### At the root of a component schema
+
+```yaml
+openapi: 3.1.0
+info:
+    title: Reference at the root of a component schema
+    version: 1.0.0
+paths:
+    /items:
+        get:
+          responses:
+            200:
+              description: ok
+              content:
+                application/json:
+                  schema:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/specialitem'
+components:
+    schemas:
+      specialitem:  # Use the item type but provide a different title for the type
+        title: Special Item
+        $ref: "#/components/schemas/item"
+      item:
+        title: Item
+        type: object
+```
+
+#### In a subschema of a component schema
+
+```yaml
+openapi: 3.1.0
+info:
+    title: Reference in a subschema of an component schema
+    version: 1.0.0
+paths:
+    /items:
+        get:
+          responses:
+            200:
+              description: ok
+              content:
+                application/json:
+                  schema:
+                    $ref: '#/components/schemas/items'
+components:
+    schemas:
+      items:
+        type: array
+        items:
+            $ref: '#/components/schemas/item'
+      item:
+        type: object
+```
+
+#### References local to a JSON Schema Resource defined by an OpenAPI Schema
+
+JSON Schema defines the concept of a JSON Schema Resource which is identified by a URI.  $ref values can be specified relative to the JSON Schema Resource.  Each OpenAPI schema object can be considered a JSON Schema Resource.
+
+In this example schema "a" is a JSON Schema resource and the reference in the "b" property of the "c" object is relative to the JSON Schema resource "a".  Unfortunately, OpenAPI 3.1 has no well defined URIs for OpenAPI Schemas. This should be resolved in OpenAPI 3.2 with the introduction of a new top level "self" property.
+
+```yaml
+components:
+schema:
+  a:
+    type:
+        - object
+        - 'null'
+    properties:
+        b:
+            type:
+                - object
+                - 'null'
+            properties:
+                c:
+                    type:
+                        - object
+                        - 'null'
+                    properties:
+                        b:
+                            $ref: '#/properties/b'
```

Comment:
this one looks odd to me, if I go "up one schema" I end up in the schema definition of b, which does not contain any b property. Maybe I'm using the wrong "resolving routine", in which case I believe we should document the routine one should apply here.